frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -126,23 +124,35 @@ class Gallery extends React.Component {
     document.removeEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
   }
 
-  buildBreadcrumbs = () => {
+  async loadContentDetails () {
+    const { state } = this
+
+    const workspaceLabel = await this.loadWorkspaceData()
+    let folderDetail
+    if (state.folderId) {
+      folderDetail = await this.loadFolderDetail(state.config.appConfig.workspaceId, state.folderId)
+    }
+    const breadcrumbsList = await this.buildBreadcrumbs(workspaceLabel, folderDetail, false)
```

Comment:
I'm not a fan of the fact that loadContentDetails handle the breadcrumbs internally.
the breadcrumbs depends on that function but should be handled outside of it